packages/block-editor/src/components/block-list/use-block-props/use-scroll-into-view.js

```diff
@@ -47,12 +47,23 @@ export function useScrollIntoView( clientId ) {
 			return;
 		}
 
-		const scrollContainer = getScrollContainer( extentNode );
+		let scrollContainer = getScrollContainer( extentNode );
 
-		// If there's no scroll container, it follows that there's no scrollbar
-		// and thus there's no need to try to scroll into view.
+		// If there's no scroll container, then we check whether the
+		// node is in an iframe. If it's in an iframe then we check
+		// if it's scrollable or not. If it's scrollable we proceed
+		// on and tell the `scrollIntoView` to scroll the iframe.
 		if ( ! scrollContainer ) {
-			return;
+			const { ownerDocument } = extentNode;
+			const isFramed = !! ownerDocument.defaultView?.frameElement;
+			const scrollingElement =
+				ownerDocument.scrollingElement || ownerDocument.body;
+			if (
+				isFramed &&
+				scrollingElement.scrollHeight > scrollingElement.clientHeight
+			) {
+				scrollContainer = ownerDocument.defaultView;
```

Comment:
Do we need all these checks? Why does `getScrollContainer( extentNode ) || extentNode.ownerDocument.defaultView` not work? I don't think it matters whether it's in a frame or not.

Reply:
Just added a missing return. The goal here is to save all the checks done by `scrollIntoView`. If we default to `getScrollContainer( extentNode ) || extentNode.ownerDocument.defaultView` then the `scrollContainer` will never be `falsy`, so `scrollIntoView` will be called every time.

Reply:
That seems fine?